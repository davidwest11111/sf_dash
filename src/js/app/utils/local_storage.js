export const ANGULAR_LOCAL_STORAGE_PREFIX = 'ngStorage-';
export const WEATHER_KEY_KEY = 'weatherKey';
export const WEATHER_UNITS_KEY = 'weatherUnits';
export const BUS_STOP_ROUTE_TAGS_KEY = 'stopRouteTags';
export const HTML_CLASS_KEY = 'htmlClass';
export const ALL_SETTINGS_KEYS = [
  WEATHER_KEY_KEY,
  WEATHER_UNITS_KEY,
  HTML_CLASS_KEY,
].reduce(
  // include legacy keys
  (fullArray, key) => ([...fullArray, key, `${ANGULAR_LOCAL_STORAGE_PREFIX}${key}`]),
  [],
);

export const getLocalStorage = key => (
  JSON.parse((
    window.localStorage.getItem(key) ||
    // legacy support
    window.localStorage.getItem(`${ANGULAR_LOCAL_STORAGE_PREFIX}${key}`)
  )) || ''
);

export const setLocalStorage = (key, val) => (
  window.localStorage.setItem(key, JSON.stringify(val))
);

export const dumpLocalStorage = () => {
  ALL_SETTINGS_KEYS.forEach(key => window.localStorage.removeItem(key));
};
