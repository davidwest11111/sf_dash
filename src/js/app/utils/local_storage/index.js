export const ANGULAR_LOCAL_STORAGE_PREFIX = 'ngStorage-';
export const WEATHER_KEY_KEY = 'weatherKey';
export const WEATHER_UNITS_KEY = 'weatherUnits';
export const BUS_STOP_ROUTE_TAGS_KEY = 'stopRouteTags';
export const HTML_CLASS_KEY = 'htmlClass';
export const ALL_SETTINGS_KEYS = [
  WEATHER_KEY_KEY,
  WEATHER_UNITS_KEY,
  BUS_STOP_ROUTE_TAGS_KEY,
  HTML_CLASS_KEY,
].reduce(
  // include legacy keys
  (fullArray, key) => ([...fullArray, key, `${ANGULAR_LOCAL_STORAGE_PREFIX}${key}`]),
  [],
);

export default class LocalStorage {
  static get(key) {
    return JSON.parse((
      window.localStorage.getItem(key) ||
      // legacy support
      window.localStorage.getItem(`${ANGULAR_LOCAL_STORAGE_PREFIX}${key}`)
    )) || '';
  }

  static set(key, val) {
    window.localStorage.setItem(key, JSON.stringify(val));
  }

  static dump() {
    ALL_SETTINGS_KEYS.forEach(key => window.localStorage.removeItem(key));
  }

  static addBusStopToLocalStorage(routeStopTag) {
    const stopRouteTags = this.get(BUS_STOP_ROUTE_TAGS_KEY) || [];
    this.set(
      BUS_STOP_ROUTE_TAGS_KEY,
      [...new Set([...stopRouteTags, routeStopTag])], // no duplicates
    );
  }

  static removeBusStopFromLocalStorage(routeStopTag) {
    const stopRouteTags = this.get(BUS_STOP_ROUTE_TAGS_KEY) || [];
    this.set(
      BUS_STOP_ROUTE_TAGS_KEY,
      stopRouteTags.filter(tag => tag !== routeStopTag),
    );
  }
}
