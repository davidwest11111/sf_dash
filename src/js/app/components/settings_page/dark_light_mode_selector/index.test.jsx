import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';

import * as LocalStorageUtils from 'app/utils/local_storage';

import DarkLightModeSelector from './index';


describe('outputs the expected tree when', () => {
  let wrapper;
  let instance;

  beforeEach(() => {
    wrapper = shallow((
      <DarkLightModeSelector />
    ));
    instance = wrapper.instance();
  });

  test('when dark mode is selected', () => {
    jest.spyOn(instance, 'isDarkModeSelected').mockReturnValue(true);
  });

  test('when light mode is selected', () => {
    jest.spyOn(instance, 'isDarkModeSelected').mockReturnValue(false);
  });

  afterEach(() => {
    instance.forceUpdate();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('instance methods', () => {
  let wrapper;
  let instance;

  beforeEach(() => {
    wrapper = shallow((
      <DarkLightModeSelector />
    ));
    instance = wrapper.instance();
  });

  test('setMode', () => {
    const mode = 'dark';
    const setLocalStorageSpy = jest.spyOn(LocalStorageUtils, 'setLocalStorage');
    const forceUpdateSpy = jest.spyOn(instance, 'forceUpdate');

    instance.setMode(mode);

    expect(setLocalStorageSpy).toHaveBeenCalledWith(LocalStorageUtils.HTML_CLASS_KEY, mode);
    expect(document.documentElement.className).toEqual(mode);
    expect(forceUpdateSpy).toHaveBeenCalled();
  });

  test('button clicks call setMode', () => {
    const spy = jest.spyOn(instance, 'setMode');

    wrapper.find('Button').at(0).simulate('click');
    expect(spy).toHaveBeenCalledWith('dark-mode');

    wrapper.find('Button').at(1).simulate('click');
    expect(spy).toHaveBeenCalledWith('light-mode');
  });
});
