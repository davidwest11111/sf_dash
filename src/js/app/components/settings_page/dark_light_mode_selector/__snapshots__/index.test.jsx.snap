// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outputs the expected tree when when dark mode is selected 1`] = `
<Fragment>
  <Label
    className="mr-2"
    tag="label"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    Theme:
  </Label>
  <ButtonGroup
    className=""
    role="group"
    tag="div"
  >
    <Button
      color="primary"
      onClick={[Function]}
      tag="button"
    >
      Dark
    </Button>
    <Button
      color="secondary"
      onClick={[Function]}
      tag="button"
    >
      Light
    </Button>
  </ButtonGroup>
</Fragment>
`;

exports[`outputs the expected tree when when light mode is selected 1`] = `
<Fragment>
  <Label
    className="mr-2"
    tag="label"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    Theme:
  </Label>
  <ButtonGroup
    className=""
    role="group"
    tag="div"
  >
    <Button
      color="secondary"
      onClick={[Function]}
      tag="button"
    >
      Dark
    </Button>
    <Button
      color="primary"
      onClick={[Function]}
      tag="button"
    >
      Light
    </Button>
  </ButtonGroup>
</Fragment>
`;
